### ISSUE TITLE
Same `crates` folder for crates.io and locally published crates

### ISSUE THREAD

**Issue by Christiaan676** [kellnr/kellnr#178]
I noticed that Kellnr uses the `crates` folder in the data directory to store all the cached crates from crates.io. Wat worries me is that it also stores the locally published crates in that directory. Doing this may lead to name clashes. 

Did I miss something in configuration or is this intended?

**Comment by Christiaan676** [comment]
Did some further testing. Creating a project tokio (`cargo new tokio`) and tried publishing a version (1.22) that is not yet cashed from crate.io and a version (1.23) that is cached from crates.io in my kellnr instance. 

Publishing version 1.23 fails:
```
   Uploading tokio v1.23.0 (/....../tokio)
error: failed to publish to registry at https://<kellnr registry URL>:443

Caused by:
  the remote server responded with an error: ERROR: Crate with version already exists: tokio-1.23.0
```

Publishing version 1.22 works just fine:
```
cargo publish --registry kellnr --allow-dirty
    Updating `kellnr` index
warning: manifest has no description, license, license-file, documentation, homepage or repository.
See https://doc.rust-lang.org/cargo/reference/manifest.html#package-metadata for more info.
   Packaging tokio v1.22.0 (/..../tokio)
   Verifying tokio v1.22.0 (/..../tokio)
   Compiling tokio v1.22.0 (/..../tokio/target/package/tokio-1.22.0)
    Finished dev [unoptimized + debuginfo] target(s) in 0.22s
    Packaged 4 files, 990.0B (737.0B compressed)
   Uploading tokio v1.22.0 (/..../tokio)
    Uploaded tokio v1.22.0 to registry `kellnr`
note: Waiting for `tokio v1.22.0` to be available at registry `kellnr`.
You may press ctrl-c to skip waiting; the crate should be available shortly.
   Published tokio v1.22.0 at registry `kellnr`
```

Now using the  tokio = "=1.22.0" in an other project, that is setup to use the kellnr as crates.io proxy will pull my version of the tokio crates. Luckily the build fails:
```
cargo build
    Updating `kellnr` index
  Downloaded tokio v1.22.0 (registry `kellnr`)
error: failed to download replaced source registry `crates-io`

Caused by:
  failed to verify the checksum of `tokio v1.22.0 (registry `kellnr`)`
```

Looking at this, my assumption is that the index is correctly separated, hence the hash miss match but the file store is not.

**Comment by Christiaan676** [comment]
Hi @secana,

As I have no experience with SeaORM so I'm fine with you creating the migration. Thing I learned is that you can't iterate over the files, as you can't decode the crate name and version from the file name (crate names and version may contain '-'). So I think you need to iterate over all the entries in the database and check is there is a matching file and move it to the correct location.

**Comment by secana** [comment]
I've added the corresponding migration on the devel branch. I'll fix a more bugs and then merge all fixes into master for a kellnr 5.1.2 release.
